R/commonsurvivalanalysis.R

```diff
@@ -67,27 +82,37 @@
       exitAnalysisIfErrors         = TRUE
     )
 
-  if (!is.null(options[["modelTerms"]]))
+  if (!is.null(subgroupVariable))
+    .hasErrors(
+      dataset                      = dataset,
+      type                         = c("observations", "variance"),
+      all.target                   = subgroupVariable,
+      observations.amount          = "< 2",
+      exitAnalysisIfErrors         = TRUE
+    )
+
+  if (!is.null(options[["modelTerms"]]) && type == "Cox")
     .hasErrors(
       dataset                      = dataset,
       type                         = c("modelInteractions"),
       modelInteractions.modelTerms = options[["modelTerms"]],
       exitAnalysisIfErrors         = TRUE
     )
 
+  if (!is.null(weightsVariable)) {
+    if (!all(isWholenumber(dataset[[weightsVariable]])))
+      .quitAnalysis(gettextf("The weights variable must be an integer."))
+    if (anyNA(dataset[[weightsVariable]]))
+      .quitAnalysis(gettextf("The weights variable must not contain missing values."))
+    if (any(dataset[[weightsVariable]] < 1))
+      .quitAnalysis(gettextf("The weights variable must be positive."))
```

Comment:
```suggestion
    if (!all(isWholenumber(dataset[[weightsVariable]])))
      .quitAnalysis(gettext("The weights variable must be an integer."))
    if (anyNA(dataset[[weightsVariable]]))
      .quitAnalysis(gettext("The weights variable must not contain missing values."))
    if (any(dataset[[weightsVariable]] < 0))
      .quitAnalysis(gettext("The weights variable must be positive."))
```
`gettextf` -> `gettext` since there is no interpolation? Also, for the third check I think you want to check `< 0` instead of `< 1`?